org.eclipse.jdt.debug/model/org/eclipse/jdt/debug/core/JDIDebugModel.java

```diff
@@ -149,6 +152,61 @@ public class JDIDebugModel {
 	 */
 	public static final String DISABLE_HCR_LAUNCH_ATTRIBUTE = getPluginIdentifier() + ".disable.hcr"; //$NON-NLS-1$
 
+	/**
+	 * The user specified a filter, can be used for highlighting specific, very important code layers.
+	 *
+	 * @since 3.22
+	 */
+	public static final Category CATEGORY_CUSTOM_FILTERED = new Category("CUSTOM_FILTERED", false); //$NON-NLS-1$
```

Comment:
As said i won't accept any particular Category as API. Can you make the List of Categories somehow non-api?

Reply:
I've removed the list of Categories. I don't think removing the String/Category constants would make sense, but if you prefer, I can move those constants from JDIDebugModel to StackFrameCategorizer for example

Reply:
When we would make something API it would stay almost forever. That should only be done for things that are so well crafted that they are unlikely to change. This Categories are not set to stone.

Reply:
What makes something into an API in your opinion? As the logic of deciding of the categories is in a different module from the code which needs those categories to actually pick the different colors, or to show the UI to configure these categories, in the end, we need to have some public constants that ensures that they are referring to the same thing.
 What do you suggest what to do?

Reply:
A API is a thing that can be accessed from another plugin. For example a public class/member in a public package.
Sorry, I don't have the time to think about a concrete solution for this PR. I can just guess, that it would be enough if either the UI or the core knows all categories at compile time and tells the other plugin at runtime what it wants.